Sources/CreateAPI/Generator/Generator+Schemas.swift

```diff
@@ -321,7 +321,18 @@ extension Generator {
         entity.properties = try makeInlineProperties(for: name, object: details, context: context)
             .filter { !$0.type.isVoid }
             .removingDuplicates(by: \.name) // Sometimes Swifty bool names create dups
-        entity.protocols = getProtocols(for: name, context: context)
+        
+        entity.protocols = {
+            var protocols = getProtocols(for: name, context: context)
```

Comment:
I had a quick scan, and it looks like `getProtocols(for:context:)` is used in four different locations since we create `EntityDeclaration`'s in a few different places. 

I tried to understand what impact that has, and it seems to be this.. Lets imagine the Petstore schema had this additional schema:

```diff
--- a/Tests/CreateAPITests/Specs/petstore.yaml
+++ b/Tests/CreateAPITests/Specs/petstore.yaml
@@ -97,6 +97,17 @@ components:
           type: string
         tag:
           type: string
+    Dog:
+      allOf:
+        - $ref: "#/components/schemas/Pet"
+        - type: object
+          properties:
+            breed:
+              type: string
+              enum:
+                - "Large"
+                - "Medium"
+                - "Small"
     Store:
       description:
       type: object
```

Because this is an `allOf` type, the entity is created differently, but when using `isInliningPropertiesFromReferencedSchemas`, it would still have an `id` and we'd therefore still want it to conform to `Identifiable` but it doesn't:

```diff
--- a/Tests/CreateAPITests/Expected/petstore-identifiable/Sources/Entities.swift
+++ b/Tests/CreateAPITests/Expected/petstore-identifiable/Sources/Entities.swift
@@ -21,6 +21,43 @@ public struct Pet: Codable, Identifiable {
     }
 }
 
+public struct Dog: Codable {
+    public var id: Int
+    /// Example: "Buddy"
+    public var name: String
+    public var tag: String?
+    public var breed: Breed?
+
+    public enum Breed: String, Codable, CaseIterable {
+        case large = "Large"
+        case medium = "Medium"
+        case small = "Small"
+    }
```

Looking at the way that the `Hashable` conformance is done too, it seems to suffer the same issue and needs to be fixed (we can do that separately 😄).

To fix it for `isGeneratingIdentifiableConformance` however, I think that we should move it inside of `getProtocols(for:context:)`. WDYT? 

I guess it means that we'll need to pass `[Property]` into the method as well, but that is fine. It will be useful for fixing the `Hashable` implementation later too (I think we can put that behind an option as well)

Reply:
That makes a lot of sense, will do!

Reply:
Thanks!

> Is there also a reason why `type` should be included as a parameter?

You're right, it looks unused so we can probably remove it 😄